conan/internal/deploy.py

```diff
@@ -156,7 +163,7 @@ def _flatten_directory(dep, src_dir, output_dir, symlinks, extension_filter=None
 
             try:
                 file_count += 1
-                shutil.copy2(src_filepath, dest_filepath, follow_symlinks=symlinks)
+                shutil.copy2(src_filepath, dest_filepath, follow_symlinks=not symlinks)
```

Comment:
This is confusing to me. Why `follow_symlinks=not symlinks)`? I'd put a comment explaining that.

Reply:
Done by the commit 5220ada30

In summary, in case the src is a link and follow_link is True, copy2 will not preserve the link, but do a copy of its target instead, so libfoo.so.1 (a symlink in package folder), will be deployed with same name, but will be a copy of libfoo.so instead.

Reply:
Done [5220ada](https://github.com/conan-io/conan/pull/17824/commits/5220ada3082608cdf6ae3eb104fd97c796437c95)